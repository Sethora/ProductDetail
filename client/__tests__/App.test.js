import React from 'react';
import { shallow } from 'enzyme';
import App from '../src/containers/app';
import Shipping from '../src/components/shipping';
import GlobalFonts from '../src/assets/fonts/fonts';
import Preview from '../src/containers/preview';
import Tabs from '../src/components/tabs';

describe('App', () => {
  let wrapper;
  console.log(shallow)
  beforeAll((done) => {
    wrapper = shallow(<App />);
    done();
  });

  test('should render correctly', () => expect(wrapper).toMatchSnapshot());
  test('should contain Shipping tag', () => expect(wrapper.contains(<Shipping />)).toBe(true));
  test('should contain the global fonts', () => expect(wrapper.contains(<GlobalFonts />)).toBe(true));
  test('should contain the Preview component', () => expect(wrapper.contains(<Preview product={{}} />)).toBe(true));
  test('should contain the Tabs component', () => expect(wrapper.contains(<Tabs tabs={[]} />)).toBe(true));
});
