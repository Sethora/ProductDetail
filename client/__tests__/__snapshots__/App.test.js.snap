// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<ModalProvider>
  <Memo(l) />
  <styled.div>
    <Preview
      product={Object {}}
    />
    <section
      id="details"
    >
      <Basket
        product={Object {}}
      />
      <Shipping />
      <Tabs
        tabs={Array []}
      />
    </section>
  </styled.div>
</ModalProvider>
`;
